src/robot/conf/gatherfailed.py

```diff
@@ -32,6 +32,21 @@ def visit_keyword(self, kw):
         pass
 
 
+class GatherFailedSuites(SuiteVisitor):
+
+    def __init__(self, critical_tags, non_critical_tags):
+        self.suites = set()
+        self._criticality = Criticality(critical_tags,
+                                        non_critical_tags)
```

Comment:
@pekkaklarck yeah I first implemented this way but had a failure with my tests:
`Processing output failed: IOError: [Errno 2] No such file or directory: u'/tmp/robottests/Python-2.7-Linux/output/output.xml'`
looking closer at the documentation of showed me I missed the necessity to only set it on the root suite :face_with_thermometer: